### DISCUSSION TITLE
Equivalent of construct() for Union types for bypassing validation

### DISCUSSION THREAD

**Discussion by Norbo11** [pydantic/pydantic#2992] [Question]
Hello, I have a set of models which look like this:

```
class InnerA(MyBaseModel):
    a: str


class InnerB(MyBaseModel):
    b: str


class Outer(MyBaseModel):
    inner: InnerA
    list_of_inner: List[InnerA]
    list_of_any_inners: List[Union[InnerA, InnerB]]
```

I want to be able to construct a full `Outer` model from a raw dictionary while skipping validation.

I discovered that I can use the `construct()` method like this:

```
outer = Outer.construct(
    inner={"a": "a"},
    list_of_inner=[{"a": "a"}],
    list_of_any_inners=[{"a": "a"}, {"b": "b"}],
)
```

The first issue I ran into is that the properties of Outer become raw dictionaries (as in raw_data), rather than being constructed into appropriate models. 

```
> outer
> Outer(inner={'a': 'a'} list_of_inner=[{'a': 'a'}] list_of_any_inners=[{'a': 'a'}, {'b': 'b'}])
```

I discovered [this issue](https://github.com/samuelcolvin/pydantic/issues/1168) which has alleviated that issue by recursively attempting to construct appropriate model instances based on the field values. However, this only works in the case of the `inner` and `list_of_inner` properties, but not for `list_of_any_inners`:

```
> outer
> Outer(inner=InnerA(a='a') list_of_inner=[InnerA(a='a')] list_of_any_inners=[{'a': 'a'}, {'b': 'b'}])
```

This is because the custom implementation of `construct` taken from the issue above does not handle constructing an appropriate subtype of a union. I was wondering whether there is a method available in the pydantic codebase somewhere which would allow me to handle this case. It must be possible since pydantic does this by default under normal operation, by matching the structure data to the schema of each subtype model. Could anyone point me to such a function, or advise how else I could get around this?

Many thanks.

**Comment by lesnik512** [comment]
Hello, which subtype of the Union should be considered appropriate without validation?)

I tried to do some experiments. So if it's Union, I choose first pydantic-model from it
```python
from pydantic import BaseModel as PydanticBaseModel
import typing

class MyBaseModel(PydanticBaseModel):
    @classmethod
    def construct(cls, _fields_set=None, **values):
        m = cls.__new__(cls)
        fields_values = {}

        config = cls.__config__

        for name, field in cls.__fields__.items():
            key = field.alias
            if key not in values and config.allow_population_by_field_name: # Added this to allow population by field name
                key = name

            if key in values: 
                if values[key] is None and not field.required: # Moved this check since None value can be passed for Optional nested field
                    fields_values[name] = field.get_default()
                else:
                    model_ = None
                    if isinstance(field.type_, typing._GenericAlias):
                        classes = [x for x in field.type_.__args__ if issubclass(x, BaseModel)]
                        if classes:
                            model_ = classes[0]
                    elif issubclass(field.type_, BaseModel):
                        model_ = field.type_
                    if model_:
                        if field.shape == 2:
                            fields_values[name] = [
                                model_.construct(**e)
                                for e in values[key]
                            ]
                        else:
                            fields_values[name] = field.outer_type_.construct(**values[key])
                    else:
                        fields_values[name] = values[key]
            elif not field.required:
                fields_values[name] = field.get_default()

        object.__setattr__(m, '__dict__', fields_values)
        if _fields_set is None:
            _fields_set = set(values.keys())
        object.__setattr__(m, '__fields_set__', _fields_set)
        m._init_private_attributes()
        return m
```
Result: 
```
Outer(inner=InnerA(a='a'), list_of_inner=[InnerA(a='a')], list_of_any_inners=[InnerA(a='a'), InnerA()])
```
So it seems like to much work to define which type from Union should be constructed.

**Reply by lesnik512** [reply]
As a partial solution in case of Union type, you can use `parse_obj_as`
